semgrep-core/typing/Typechecking_generic.ml

```diff
@@ -46,6 +46,15 @@ let compatible_type t e =
     L (Float _) -> true
   | OtherType (OT_Expr, [E (Id (("str", _tok), _idinfo))]),
     L (String _) -> true
+  | TyBuiltin ((t1, _)), Id (_, {id_type; _}) ->
```

Comment:
sounds good for a v0.
You can add a comment here saying we should handled type metavariables and do more general
type matching (use m_type in Generic_vs_generic)